util/merkle/src/lib.rs

```diff
@@ -66,6 +66,26 @@ impl fmt::Display for TrieError {
 /// Trie result type. Boxed to avoid copying around extra space for `H256`s on successful queries.
 pub type Result<T> = ::std::result::Result<T, Box<TrieError>>;
 
+/// Description of what kind of query will be made to the trie.
+pub trait Query {
```

Comment:
Why do you need this trait?
Isn't `Decodable` doing the job for this kind of property?

Reply:
It's used in `get_with` method. A `get_with` is more flexible than just returning the decoded value.
It receives the bytes and returning any type that can be made by the bytes.

I think this flexibility is more than we want, and we can refactor it to return the specified type.
But it should be another patch, since replacing implementation and interface in one commit makes review hard.

Reply:
Confirmed.